### ISSUE TITLE
[Feature Request] Check for new release

### ISSUE THREAD

**Issue by borgmanJeremy** [flameshot-org/flameshot#1237]
Add a feature that lets flameshot query the github release API to determine if it has the latest version. Useful for windows and Linux.

@mmahmoudian I discovered the team at namecheap(@thepurple) added this feature already. Do you see any privacy concerns with this? I am not sure if I should leave it in during the merge or pull it out. I think it will be pretty nice for appimage, MacOs, and Linux but I know some people hate anything that can phone home.

**Comment by mmahmoudian** [comment]
I have used the API so far through `curl` in my scripts to get the release page and then grep to extract the latest version:

```
curl -O -J -L $(curl -s https://api.github.com/repos/flameshot-org/flameshot/releases/latest \
               | grep -Po 'https://github.com/flameshot-org/flameshot/releases/download/[^}]*\.AppImage' \
               | uniq)
```

I don't see any privacy issues with this as the only information sent to Github are

- IP address (can be mitigated through Tor, proxy, VPN)
- interest in the target repo (not a big deal imho)
- And if we implement a download button, then it also reveals to Github what OS the user uses (again imho not a big deal imho)

To sum up, if the API you are talking about is the same API that I used in the code snippet above, I don't see and privacy issue as browsing the web in normal way reveals much more information to Github including your browser, screen size, locale, etc. As demonstrated in EFF:

https://coveryourtracks.eff.org/

**Comment by borgmanJeremy** [comment]
Awesome, I think this is a great feature. Will get merged soon.
